### DISCUSSION TITLE
Update/Upgrade management for the consumers

### DISCUSSION THREAD

**Discussion by amir-ziaei** [epicweb-dev/epic-stack#21] [Ideas]
Hey everyone,

First of all, I must say The Epic Stack looks awesome! Thanks Kent and all other contributors.

Secondly, I'd like to raise a concern that I believe is not unique to this stack, but is also applicable to other heavily opinionated stacks. I'm interesting in hearing your thoughts and opinions on it.

As you're aware, adopting this stack introduces a significant number of dependencies for developers to maintain in their projects. This is understandable considering the benefits it offers. However, I believe the stack itself has to maintain these dependencies, and even more so, to ensure that newer project scaffolds include the latest stable goodies.

It seems like a similar task will be performed in different repositories. Maintainers here be maintaining the dependencies, while also people be maintaining the same dependencies over in their apps. Perhaps we could find a way to make this process more efficient for the consumers.

I'm not sure how feasible it would be to provide a mechanism that seamlessly updates dependencies for those who choose to use the stack. Nonetheless, as a consumer, it would be a compelling selling point if I knew I didn't have to worry about manually updating these dependencies or periodically revisiting the repository to sync my own repo with the latest opinions and patches.

While this would be fantastic for consumers, I realize it could pose challenges for the stack maintainers. Swapping dependencies would become more complicated since they would need to consider the impact on people's apps. There may be other challenges that I haven't considered as well.

In any case, I would appreciate your thoughts on this matter. Feel free to let me know if I'm being overly optimistic or if there is already a solution to this that I'm unaware of 😃.

Thank you!

**Comment by kentcdodds** [comment]
Hi Amir! Thanks for looking into the stack!

I don't think there's any good way to help people keep their dependencies up-to-date. Though I did do something for development deps when I was at PayPal and for my open source stuff with kcd-scripts. Perhaps we could do something like that in the future.

For now, I think it's best to just let each project maintain its own dependencies.

I also hope that much of the code in this project can be extracted and turned into libraries so people don't have to worry about keeping the code itself up-to-date as new approaches come up.

In any case, I think I'll resolve this discussion since I don't think there's anything actionable right now, but maybe one day. Thanks!
